students/CarolF/module05/except_exercise_W5_CF.py

```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/python
+
+"""
+###Edited and submitted by Carol Farris
+##Week 05 Exceptions practice
+
+An exercise in playing with Exceptions.
+Make lots of try/except blocks for fun and profit.
+
+Make sure to catch specifically the error you find, rather than all errors.
+"""
+
+from except_test import fun, more_fun, last_fun
+
+
+# Figure out what the exception is, catch it and while still
+# in that catch block, try again with the second item in the list
+first_try = ['spam', 'cheese', 'mr death']
+
+
+try:
+    joke = fun(first_try[0])
+except NameError:
+	joke = fun(first_try[1])
+
+
+# Here is a try/except block. Add an else that prints not_joke
+try:
+    not_joke = fun(first_try[2])
+except SyntaxError:
+    print('Run Away!')
+else:
+    print(not_joke)    
+
+# What did that do? You can think of else in this context, as well as in
+# loops as meaning: "else if nothing went wrong"
+# (no breaks in  loops, no exceptions in try blocks)
+
+# Figure out what the exception is, catch it and in that same block
+#
+# try calling the more_fun function with the 2nd language in the list,
+# again assigning it to next_joke.
+#
+# If there are no exceptions, call the more_fun function with the last
+# language in the list regardless of whether there was an exception
+
+# Finally, while still in the try/except block and regardless of whether
+# there were any exceptions, call the function last_fun with no
+# parameters. (pun intended)
+
+langs = ['java', 'c', 'python']
+
+try:
+    more_joke = more_fun(langs[0])
+except IndexError:
+	next_joke = more_fun(langs[1])
+
+
+more_fun(langs[2])
```

Comment:
more_fun should be in an else block of the try/except, so it is only run if nothing goes wrong. last_fun we intended you to put into a finally block in that same try/except block, so that is run even if there is an error.